core/shared/src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -1,46 +1,88 @@
 package com.monovore.decline
 
-import java.net.{URI, URISyntaxException}
-
 import cats.data.{Validated, ValidatedNel}
+import java.net.{URI, URISyntaxException, URL, MalformedURLException}
+import java.nio.file.{FileSystems, InvalidPathException, Path}
+import java.util.UUID
+import scala.concurrent.duration.Duration
 
 trait Argument[A] { self =>
-  def defaultMetavar: String
   def read(string: String): ValidatedNel[String, A]
 }
 
 object Argument extends PlatformArguments {
 
   def apply[A](implicit argument: Argument[A]) = argument
 
-  implicit val readString: Argument[String] = new Argument[String] {
+  def instance[A](f: String => ValidatedNel[String, A]): Argument[A] =
+    new Argument[A] {
+      def read(s: String): ValidatedNel[String, A] = f(s)
+    }
 
-    override def read(string: String): ValidatedNel[String, String] = Validated.valid(string)
+  implicit val readString: Argument[String] =
+    instance(s => Validated.valid(s))
 
-    override def defaultMetavar: String = "string"
-  }
+  private def readNum[A](typeName: String)(parse: String => A): Argument[A] =
+    instance { s =>
+      try Validated.valid(parse(s))
+      catch { case nfe: NumberFormatException => Validated.invalidNel(s"Invalid $typeName: $s") }
+    }
 
-  private def readNum[A](typeName: String)(parse: String => A): Argument[A] = new Argument[A] {
-    override def read(string: String): ValidatedNel[String, A] =
-      try Validated.valid(parse(string))
-      catch { case nfe: NumberFormatException => Validated.invalidNel(s"Invalid $typeName: $string") }
+  implicit val readBoolean: Argument[Boolean] =
```

Comment:
I'd originally left this out since it leads to pretty UNIX-unidiomatic CLIs... instead of `cmd --foo true` or `cmd --foo false`, people usually do `cmd --foo` and `cmd --no-foo` or similar.